### ISSUE TITLE
Are inversion statements produced by Gridss incomplete?

### ISSUE THREAD

**Issue by johanneskoester** [PapenfussLab/gridss#378]
Gridss reports inversions as e.g. 
```
1    7602360    gridss0ff_396o    A    A]1:50381040]    60.52    LOW_QUAL;NO_ASSEMBLY    AS=0;ASC=1X399N1X;ASQ=0;ASRP=0;ASSR=0;BA=0;BANRP=0;BANRPQ=0;BANSR=0;BANSRQ=0;BAQ=0;BASRP=0;BASSR=0;BMQ=34;BMQN=20;BMQX=60;BQ=57.41;BSC=3;BSCQ=57.41;BUM=0;BUMQ=0;BVF=1;CAS=0;CASQ=0;CIPOS=-200,200;CIRPOS=-63,64;CQ=60.52;EVENT=gridss0ff_396;IC=0;IMPRECISE;IQ=0;MATEID=gridss0ff_396h;MQ=42.67;MQN=20;MQX=60;RAS=0;RASQ=0;REF=16;REFPAIR=3;RP=3;RPQ=60.52;SB=1;SC=43M3D139M1X399N1X;SR=0;SRQ=0;SVLEN=42778679;SVTYPE=BND;VF=3;SIMPLE_TYPE=INV    GT:ASQ:ASRP:ASSR:BANRP:BANRPQ:BANSR:BANSRQ:BAQ:BASRP:BASSR:BQ:BSC:BSCQ:BUM:BUMQ:BVF:CASQ:IC:IQ:QUAL:RASQ:REF:REFPAIR:RP:RPQ:SR:SRQ:VF    .:0:0:0:0:0:0:0:0:0:0:57.41:3:57.41:0:0:1:0:0:0:60.52:0:16:3:3:60.52:0:0:3
1    50381040    gridss0ff_396h    C    C]1:7602360]    60.52    LOW_QUAL;NO_ASSEMBLYAS=0;ASC=1X126N1X;ASQ=0;ASRP=0;ASSR=0;BA=0;BANRP=3;BANRPQ=60.52;BANSR=0;BANSRQ=0;BAQ=0;BASRP=0;BASSR=0;BMQ=55;BMQN=40;BMQX=60;BQ=66.34;BSC=4;BSCQ=66.34;BUM=0;BUMQ=0;BVF=3;CAS=0;CASQ=0;CIPOS=-63,64;CIRPOS=-200,200;CQ=60.52;EVENT=gridss0ff_396;IC=0;IMPRECISE;IQ=0;MATEID=gridss0ff_396o;MQ=42.67;MQN=20;MQX=60;RAS=0;RASQ=0;REF=20;REFPAIR=5;RP=3;RPQ=60.52;SB=0.5;SC=155M158D75M24D75M1X126N1X;SR=0;SRQ=0;SVLEN=42778679;SVTYPE=BND;VF=3;SIMPLE_TYPE=INV    GT:ASQ:ASRP:ASSR:BANRP:BANRPQ:BANSR:BANSRQ:BAQ:BASRP:BASSR:BQ:BSC:BSCQ:BUM:BUMQ:BVF:CASQ:IC:IQ:QUAL:RASQ:REF:REFPAIR:RP:RPQ:SR:SRQ:VF    .:0:0:0:3:60.52:0:0:0:0:0:66.34:4:66.34:0:0:3:0:0:0:60.52:0:20:5:3:60.52:0:0:3
```
I wonder what is meant by this. As can be seen from the VCF spec, an inversion in the traditional sense (inverting a piece of DNA inside a chromosome), has to be described with four breakends, see [VCF spec](https://samtools.github.io/hts-specs/VCFv4.3.pdf), section 5.4.7. If two of those are missing like in the example above, when constructing the resulting sequence, I end up having an inversion of the entire part of the chromosome right of the first breakend, see here (I first apply the left breakend, showing the resulting (incomplete) sequence, and then the right one, showing the final sequence):

![inversion](https://user-images.githubusercontent.com/1858646/90020751-94619780-dcb0-11ea-86cd-de24f2615abe.png)

1. Is that intentional?
2. If it is intentional, is gridss also able to report traditional smaller inversions as one event of four records?
3. If it is intentional, shouldn't there be a single breakend denoting that the reference ends left of 7602361, because that would be the new end of the chromosome (the lower right green end on my drawing)?

Thanks a lot for your help!

**Comment by d-cameron** [comment]
>Is that intentional?

Fundamentally, GRIDSS is a breakpoint caller and reports breakpoints. `simple-event-annotation.R` is intentionally not included as part of offical GRIDSS releases as GRIDSS does not do variant interpretation. A `SIMPLE_TYPE=INV` call with 


> If it is intentional, is gridss also able to report traditional smaller inversions as one event of four records?

If there is a true inversion, GRIDSS will report 2 breakpoints (4 breakends). In practice, `INV` isn't actually a very useful.

> If it is intentional, shouldn't there be a single breakend denoting that the reference ends left of 7602361, because that would be the new end of the chromosome (the lower right green end on my drawing)?

No. GRIDSS merely reports breakpoints. If it can't find a single breakend at that position then it's not going to report one. Leaving aside the fact that it's almost certainly a false positive since the QUAL is terrible and there's no assembly, your example breakpoiont is not necessarily an inversion event. It could be a fold-back inversion as part of cycle of breakage-fusion-bridge event. It could be part of a chromothriptic rearrangement. Determining what sort of variant it requires a wholistic look at both the breakpoint calls and copy number calls. GRIDSS does not do this. It identifies breakpoints (and single breakends).

One of the issues with VCF (and the approach to SVs by the community in general) is the lack of distinction between variant detection/identification, and variant interpretation. A deletion requires a breakpoint in the correct orientation but it also requires a copy number loss in the 'deleted' region. If that doesn't exist, then it's not actually a deletion, it's something else (e.g. cut+paste translocation). I've been drafting many SV changes to go into VCFv4.4 and this includes one that makes the distinction between variant detection, and variant interpretation explicit.

To properly classify variants (even simple ones), we need a tool that will take breakpoints(/single breakends) and (allele specific) copy number segments, and identify rearrangement events from this. If your data is somatic data, then you're in luck as we've already made such a tool: https://www.biorxiv.org/content/10.1101/781013v1 https://github.com/hartwigmedical/hmftools/tree/master/sv-linx . As you can see from the readme, rearrangement intepretation is actually quite complicated - even when there's only 2 breakpoints involved (e.g. 'clean' inversions where the two breakpoint are at the same position on both sides are very rare in real data).

TLDR: GRIDSS is, and will remain, a breakpoint/breakend variant detection/identification tool. Variant interpretation is done downstream.

**Comment by johanneskoester** [comment]
Thanks a lot! This was very helpful!

**Comment by johanneskoester** [comment]
I was indeed misled by the INV annotation via the script, which made me think that the two records were intended as a true inversion.

**Comment by d-cameron** [comment]
> misled by the INV annotation

That's why that script is not part of official GRIDSS releases. It's only included because I got so many request for 'simple' annotations - most likely by people familiar with Manta and DELLY which both use this 'fake' INV syntax.
